tests/test-exoskeleton-rule_validation.php

```diff
@@ -0,0 +1,210 @@
+<?php
+/**
+ * Class ExoskeletonTest
+ *
+ * @package Exoskeleton
+ */
+
+/**
+ * Sample test case.
+ */
+class ExoskeletonRuleValidationTest extends WP_UnitTestCase {
+
+	/**
+	 * Pre test setup
+	 */
+	function setUp() {
+			parent::setUp();
+			// Clear existing rules.
+			$instance = Exoskeleton::get_instance();
+			$instance->rules = [];
+	}
+
+
+	/**
+	 * Check that exoskeleton validates rules with an invalid rule
+	 *
+	 * @dataProvider validRuleProvider
```

Comment:
Best Practices is to use datProviders for multiple test points. Using them for a single item is better suited for a helper/util function so that it is easier to understand the flow.